SegmentEditorDrawTube/SegmentEditorDrawTubeLib/SegmentEditorEffect.py

```diff
@@ -193,6 +201,10 @@ def onRadiusChanged(self, radius):
     self.logic.radius = radius
     self.updateModelFromSegmentMarkupNode()
 
+  def onSegmentsChanged(self, segments):
+    self.logic.segments = segments

```

Comment:
You need to store all parameters (such as numberOfLineSegments) in self.scriptedEffect node (update setMRMLDefaults, updateGUIFromMRML, updateMRMLFromGUI) and the logic should read the parameters from the node.